### PR TITLE
Add InstanceDetails migration to upgradeKyma operation

### PR DESCRIPTION
**Pull request by ksputo**
**Description**

Changes proposed in this pull request:

- Added migration for `InstanceDetails` field from latest `ProvisioningOperation` to `UpgradeKymaOperation`

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/migrations/instancedetails.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			m.log.Fatal("Failed to generate json", err)
+		}
+		m.log.WithField("operationID", op.Operation.ID).Infof("Instance details: %s", string(pretty))
+		if op.InstanceDetails.RuntimeID != "" {
+			m.log.Infof("InstanceDetails were found in operation %s, skipping", op.Operation.ID)
+			continue
+		}
+		lastProvOp, err := m.operations.GetProvisioningOperationByInstanceID(op.InstanceID)
+		if err != nil {
+			return errors.Wrap(err, "while listing operations")
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/migrations/instancedetails.go:44]: while getting provisioning operation

File: components/kyma-environment-broker/internal/migrations/instancedetails.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func (m *InstanceDetailsMigration) Migrate() error {
+	upgradeOperations, err := m.operations.ListUpgradeKymaOperations()
+	if err != nil {
+		return errors.Wrap(err, "while listing operations")
+	}
+	m.log.Infof("Performing instance details migration of %d operations", len(upgradeOperations))
+
+	for _, op := range upgradeOperations {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/migrations/instancedetails.go:31]: you can create a logger like: ``` logger := m.lo.WithField("UpgradeKymaOperation", op.Operation.ID)``` then use it

**Comment by piotrmiskiewicz** [comment]
it looks ok for me

**Comment by ebensom** [comment]
This PR is needed for https://github.com/kyma-project/control-plane/pull/434 and hence it's a blocker for the upgrade to 1.18.0 on Kyma Runtimes, as AVS  have to be handled properly.

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**